plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -198,8 +198,44 @@ public function __construct( $args = array() ) {
 		// TODO: args to be implemented.
 		unset( $this->args['customer_note'], $this->args['name'] );
 
+		$require_limit_data = ( $this->arg_isset( 'limit' ) || $this->arg_isset( 'page' ) || $this->arg_isset( 'offset' ) );
+
 		$this->build_query();
-		$this->run_query();
+
+		/**
+		 * Filters the orders array before the query takes place.
+		 *
+		 * Return a non-null value to bypass the HPOS default order queries.
+		 *
+		 * If the query includes limits via the `limit`, `page`, or `offset` arguments, we
+		 * require the `found_orders` and `max_num_pages` properties to also be set.
+		 *
+		 * @since 8.2.0
+		 *
+		 * @param array $order_data {
+		 *     An array of order data.
```

Comment:
> The first @param will always be an array

Well, by default, the value of the first param is *null*—so, at the risk of sounding pedantic, it will not always be an array. Also, leaving the value as *null* is a valid choice for any callbacks that are registered (they may only transform it to an array if some condition is met).

I think what I'm proposing is the more accurate description of the param, but, you can treat this as optional (... and, I think you're right: if the entire comment is read, things are sufficiently clear).

Reply:
> by default, the value of the first param is null—so, at the risk of sounding pedantic, it will not always be an array

Thanks for catching that, of course you're absolutely right. And I don't think that's pedantic, I think we should be exact in our words so thanks a lot for that. ☺️